### ISSUE TITLE
No resource or method named `yum_repository' for `ChefCompat::Recipe "yum"'

### ISSUE THREAD

**Issue by artem-sidorenko** [chef-cookbooks/compat_resource#122]
### Cookbook version
`12.16.1`

### Chef-client version
`12.6.0`

### Platform Details
Bento box `centos-7.2`

### Scenario:
I try to use compat_resource in the cookbook using the `yum_repository` resource

### Steps to Reproduce:
 - Add `depends 'compat_resource', '>= 12.16.1'` to the metadata.rb
 - Add `require_chef_omnibus: 12.6.0` to the provisioner in the `.kitchen.yml`
 - Invoke `yum_repository` in our recipe

### Expected Result:
Successfully converge run

### Actual Result:
```
       ================================================================================
       Recipe Compile Error in /tmp/kitchen/cookbooks/os-hardening/recipes/default.rb
       ================================================================================
       
       NoMethodError
       -------------
       No resource or method named `yum_repository' for `ChefCompat::Recipe "yum"'
       
       Cookbook Trace:
       ---------------
         /tmp/kitchen/cookbooks/os-hardening/recipes/yum.rb:46:in `block in from_file'
         /tmp/kitchen/cookbooks/os-hardening/recipes/yum.rb:45:in `each'
         /tmp/kitchen/cookbooks/os-hardening/recipes/yum.rb:45:in `from_file'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:347:in `load_recipe'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:303:in `block in include_recipe'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:302:in `each'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:302:in `include_recipe'
         /tmp/kitchen/cookbooks/os-hardening/recipes/packages.rb:30:in `from_file'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:347:in `load_recipe'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:303:in `block in include_recipe'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:302:in `each'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:302:in `include_recipe'
         /tmp/kitchen/cookbooks/os-hardening/recipes/default.rb:22:in `from_file'
         /tmp/kitchen/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/run_context.rb:347:in `load_recipe'
       
       Relevant File Content:
       ----------------------
       /tmp/kitchen/cookbooks/os-hardening/recipes/yum.rb:
       
        39:    action :run
        40:  end
        41:  
        42:  if node['os-hardening']['security']['packages']['clean']
        43:  
        44:    # remove unused repos
        45:    %w(CentOS-Debuginfo CentOS-Media CentOS-Vault).each do |repo|
        46>>     yum_repository repo do
        47:        action :remove
        48:      end
        49:    end
        50:  
        51:    # remove packages
        52:    node['os-hardening']['security']['packages']['list'].each do |pkg|
        53:      yum_package pkg do
        54:        action :purge
        55:      end
```

I'm not sure if I miss something or I do something wrong (I simply followed the steps in the README)

**Comment by SharpEdgeMarshall** [comment]
It's solved? i'm having this issue and it's critical, it's possible to force previous version of this cookbook even if it's a sub dependency?

**Comment by artem-sidorenko** [comment]
@SharpEdgeMarshall I did not have time for a deeper look :-(

You can try to use the previous version via specifying the dependency version in your metadata.rb. 
My problem is I need this functionality on the older chef versions, so using the older version of this cookbook isn't really an option

**Comment by lamont-granquist** [comment]
confirmed that this is broken on  12.13.37

**Comment by lamont-granquist** [comment]
fixed by #125 + #126 + #127
